frontends/coreir_encoder.cpp

```diff
@@ -0,0 +1,331 @@
+#include "frontends/coreir_encoder.h"
+#include "utils/logger.h"
+
+#include <iostream>
+#include <set>
+#include <sstream>
+
+using namespace CoreIR;
+using namespace smt;
+using namespace std;
+
+// helpers
+
+bool instance_of(CoreIR::Instance * inst, std::string ns, std::string name)
+{
+  auto mod = inst->getModuleRef();
+  if (!mod->isGenerated()) {
```

Comment:
Any module from corebit will not be generated, so this function will not work as intended.

Reply:
Interesting, so is the right thing to do not to check `isGenerated` at all?

Reply:
Well, it is a bit annoying due to the fact that "coreir" (aka BitVector) has a generator parameter of "width" whereas "corebit" (aka Bool) does not have any generator parameters. So the easiest for this code is to just have a function that checks if an instance is a primitive. This can be accomplished by just checking if the namespace associated with the module is either "coreir" or "corebit".